src/main/java/org/betonquest/discordbot/modules/support/ThreadUpdateListener.java

```diff
@@ -0,0 +1,74 @@
+package org.betonquest.discordbot.modules.support;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.entities.channel.forums.ForumTag;
+import net.dv8tion.jda.api.events.channel.update.ChannelUpdateAppliedTagsEvent;
+import net.dv8tion.jda.api.events.channel.update.ChannelUpdateArchivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.betonquest.discordbot.config.BetonBotConfig;
+import org.betonquest.discordbot.modules.ForumTagHolder;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * This listener sorts and adds tags when Users change them.
+ * It also closes Forum Posts when they are solved and inactive.
+ */
+public class ThreadUpdateListener extends ListenerAdapter {
+
+    /**
+     * The {@link BetonBotConfig} instance.
+     */
+    private final BetonBotConfig config;
+
+    /**
+     * Create a new {@link ThreadUpdateListener}
+     *
+     * @param api    the {@link JDA} instance
+     * @param config the {@link BetonBotConfig} instance
+     */
+    public ThreadUpdateListener(final JDA api, final BetonBotConfig config) {
+        super();
+        this.config = config;
+        api.addEventListener(this);
+    }
+
+    @Override
+    public void onChannelUpdateAppliedTags(@NotNull final ChannelUpdateAppliedTagsEvent event) {
+        final ThreadChannel channel = event.getChannel().asThreadChannel();
```

Comment:
We should also check here for the right channel like in line 55, otherwise channels will get the `setAutoArchiveDuration(ThreadChannel.AutoArchiveDuration.TIME_1_HOUR)` call, that are actually no support channel